MPC.py

```diff
@@ -172,3 +172,5 @@ def update(frame):
 animation.save('trajectory_animation2.mp4', writer='ffmpeg', fps=60)  
 
 plt.show()
+
```

Comment:
lines 10-15: Please tidy up the high-level/file-level comments to make sure they look grammatically correct

general: use comments as a way to clarify your code to the reader. For example, I use comments to separate between thematically different regions of code within a file or a code block or even a loop.

lines 150 - 172: place plotting code within its own function; also, place other easily encapsulated code within its own functions. Functions should ideally be relatively short blocks of code, but for now focus on using functions for more of your code, as you do in other files with conceptually simple code functionality.

general: Use if __name__ == "__main__" to run the main script - see https://realpython.com/if-name-main-python/ for details.

This is a quick list of things you can start with. There's always more, but code review is an ongoing process that helps us make our code better over time